### DISCUSSION TITLE
globbing in lsearch is off by default

### DISCUSSION THREAD

**Discussion by radlyeel** [msteveb/jimtcl#302] [Ideas]
This may belong in a different forum ...

I've recently decided to add Tcl to my toolkit, and combined that with trying out Jim on an embedded project.  Blending the two, I'm using jimsh (v. 0.82, built on 4/13/2024) instead of tclsh.   Following the tutorial at https://www.tcl.tk/man/tcl8.5/tutorial/tcltutorial.html, the first hiccup came on lesson 19 , where lsearch is covered.  Using jimsh, the lsearch command requires the -glob option  for the example code to work, whereas with tclsh (v.8.6) the -glob option is on by default.  I know Jim is heavily "stripped down"; was this an intentional design decision?  If so, is there a global option I can "set and forget"?  Am I overlooking something in my naiveté?  Here's a command line sequence  that shows my concern:

```
$ cat demo.tcl
set lst {aaa bbbb ccc dd}
set b_ndx [lsearch $lst b*]
puts "b* is in $lst at position $b_ndx"

$ tclsh demo.tcl
b* is in aaa bbbb ccc dd at position 1
$ jimsh demo.tcl
b* is in aaa bbbb ccc dd at position -1
```
Inserting the -glob option resulted in identical output from tclsh and jimsh.

Assuming this is intentional (and I can think of several reasons it might be), is there somewhere a consolidated list of such differences (such as an Errata page for a book, even though this would not be "errata")

**Comment by msteveb** [comment]
Honestly I can't remember the original reason for the lsearch difference. Probably just seemed more sensible. Remember that in the early days Jim Tcl was nowhere near compatible with Tcl (no regular I/O, no array, file, regexp, etc.) so exact compatibility was less important. Now I generally keep compatibility where possible.

There is no explicit list I'm aware of other than what is listed in the introduction under "Some notable differences with Tcl 8.5/8.6/8.7". The rest are in the docs, typically prefixed by "Note"

Feel free to start a wiki page

**Reply by radlyeel** [reply]
When I begin learning a language from scratch,It's my custom to gloss over topics like "Some notable differences with Tcl 8.5/8.6/8.7" since I don't have any habits left over from previous versions.  By the time I needed it, I'd forgotten it was there.  Thanks!  Now I guess I should spend some quality time on the Wiki.

**Comment by msteveb** [comment]
btw, you can always do something like this to modify builtin commands
```
local proc lsearch {args} {
  # Not 100% robust and ignores -regexp
  if {"-glob" ni $args && "-exact" ni $args} {
    set args [list -glob {*}$args]
  }
  upcall lsearch {*}$args
}
```

**Reply by radlyeel** [reply]
Thanks!
